### PR TITLE
AAP-23850: Added info. about Red Hat and Lightspeed authentication options

### PR DESCRIPTION
**Pull request by sayjadha**
This PR addresses JIRA https://issues.redhat.com/browse/AAP-23850.

Changes made:

- Added authentication options for both Red Hat and Ansible Lightspeed.
- Added corner case about onprem dev spaces for Red Hat sign-in option.

### CODE REVIEW

File: lightspeed/modules/proc_login-vscode-extension.adoc [hunk 1]
```diff
@@ -5,7 +5,34 @@
...
+
+* *Ansible Lightspeed on-premise deployments*
++
+Users are authenticated using your {PlatformName} login. 
++
+To sign in, you can use the *Connect* button or the *Accounts* button, and use the option *Sign in with Ansible Lightspeed to use Ansible* from VS Code. Once prompted, select *Log in with {PlatformNameShort}*, and log in with the authorization mechanism that your automation controller is configured with. 
+
+* *Ansible Lightspeed cloud service*
++
+Users are authenticated using Red Hat Single Sign-On (RH-SSO).
```
Comment (hherbly) [inline lightspeed/modules/proc_login-vscode-extension.adoc:20]: There's an attribute for Red Hat Single Sign-On and for RH-SSO. Respectively, they are {RHSSO} and {RHSSOshort}

**Review by hherbly — APPROVED** [review]
LGTM

File: lightspeed/modules/proc_login-vscode-extension.adoc [hunk 2]
```diff
@@ -5,7 +5,34 @@
...
++
+To sign in, you can use the *Connect* button or the *Accounts* button and then select one of the following options:
+
+** *Sign in with Ansible Lightspeed to use Ansible*
+** *Sign in with Red Hat to use Ansible*
```
Comment (omaciel) [inline lightspeed/modules/proc_login-vscode-extension.adoc:25]: Should this also use `{RHSSO} ` or is it explicitly using `Red Hat`?

File: lightspeed/modules/proc_login-vscode-extension.adoc [hunk 3]
```diff
@@ -5,7 +5,34 @@
...
++
+Based on the option you selected, follow the on-screen prompts to log in and access the Ansible Lightspeed service.
++
+[NOTE]
++
+====
+If you are using a cloud development environment at a domain unknown by Ansible Lightspeed, such as on-premise Red Hat OpenShift Dev Spaces, your *Accounts* sign-in menu provides the option *Sign-in with Red Hat to use Ansible*. This option uses a device code flow to successfully complete the sign-in process and requires the Red Hat Authentication extension v0.2.0 or later. If you require this authentication flow but don't see the *Sign-in with Red Hat to use Ansible* option, ensure you are using the Ansible VS Code extension v24.5.2 or later.
```
Comment (omaciel) [inline lightspeed/modules/proc_login-vscode-extension.adoc:32]: Same question as before about using `{RHSSO} `

File: lightspeed/modules/proc_login-vscode-extension.adoc (hunk 2 above)
Comment (sayjadha) [reply lightspeed/modules/proc_login-vscode-extension.adoc:25]: This is the UI label

File: lightspeed/modules/proc_login-vscode-extension.adoc (hunk 3 above)
Comment (sayjadha) [reply lightspeed/modules/proc_login-vscode-extension.adoc:32]: This is the UI label

**Review by hherbly — APPROVED** [review]

**Pull request merged**